File: source/ResourceLibrary/Models/View/LibraryViewModel.vb

Imports Contensive.BaseClasses


Namespace Models.View
    Public Class LibraryViewModel
        Inherits DesignBlockViewBaseModel
        '
        Public Property bodyHtml As String
        '
        '====================================================================================================
        ''' <summary>
        ''' Populate the view model from the entity model
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="settings"></param>
        ''' <returns></returns>
        Public Overloads Shared Function create(cp As CPBaseClass, settings As Models.Db.ResourceLibraryModel, htmlBody As String) As LibraryViewModel
            Try
                '
                ' -- base fields
                Dim result = DesignBlockViewBaseModel.create(Of LibraryViewModel)(cp, settings)
                '
                ' -- custom
                result.bodyHtml = htmlBody
                '
                Return result
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
                Return Nothing
            End Try
        End Function
    End Class

End Namespace

File: source/ResourceLibrary/Models/View/LibraryViewModel.cs

using System;
using Contensive.BaseClasses;

namespace Contensive.Addons.ResourceLibrary.Models.View {
    public class LibraryViewModel : DesignBlockViewBaseModel {
        public string bodyHtml { get; set; }
        public static LibraryViewModel create(CPBaseClass cp, Models.Db.ResourceLibraryModel settings, string htmlBody) {
            try {
                var result = DesignBlockViewBaseModel.create<LibraryViewModel>(cp, settings);
                result.bodyHtml = htmlBody;
                return result;
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
                return null;
            }
        }
    }
}
